src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs

```diff
@@ -360,19 +360,19 @@ protected static ILogger CreateLogger<T>(ILoggerFactory factory)
 
         protected static async Task WriteFileAsync(HttpContext context, Stream fileStream, RangeItemHeaderValue range, long rangeLength)
         {
-            var outputStream = context.Response.Body;
-            using (fileStream)
+            var outputPipeWriter = context.Response.BodyWriter;
+            await using (fileStream)
             {
                 try
                 {
                     if (range == null)
                     {
-                        await StreamCopyOperation.CopyToAsync(fileStream, outputStream, count: null, bufferSize: BufferSize, cancel: context.RequestAborted);
+                        await StreamCopyOperation.CopyToAsync(fileStream, outputPipeWriter, count: null, cancel: context.RequestAborted);
                     }
                     else
                     {
                         fileStream.Seek(range.From.Value, SeekOrigin.Begin);
```

Comment:
Does here also need a check for `range.From.Value > 0` before seek?
Also does it need a null check for `range.From`?